torch/testing/_internal/distributed/rpc_utils.py

```diff
@@ -37,6 +42,20 @@
 )
 from torch.testing._internal.distributed.rpc.examples.parameter_server_test import ParameterServerTest
 
+def _check_and_set_tcp_init():
+    # if we are running with TCP init, set main address and port
+    # before spawning subprocesses, since different processes could find
+    # different ports.
+    use_tcp_init = os.environ.get("RPC_INIT_WITH_TCP", None)
+    if use_tcp_init== "1":
+        os.environ["MASTER_ADDR"] = '127.0.0.1'
+        os.environ["MASTER_PORT"] = str(find_free_port())
```

Comment:
As I understand it, this actually may not be enough to prevent port collisions if there are multiple tests running in parallel? Test 1 could get port X, but before it creates the store and binds the port, Test 2 could also get port X and we would still have the race. If my understanding is correct, we probably need to write currently in use ports to a shared/locked data structure to completely avoid races?

Reply:
(This probably won't be an issue anymore now that we are not enabling it for all tests)